### DISCUSSION TITLE
Fourth option for focusRing?

### DISCUSSION THREAD

**Discussion by Nantris** [mantinedev/mantine#2928] [Feedback]
There seems to be a gap in  the `focusRing` logic that leaves programmatically-focused elements out of having the `focusRing` styles applied to them.

Right now the only option is to use `focusRing: 'always'`, but it makes for an ugly experience when the user is clicking with a mouse.

But using `focusRing: 'auto'` leaves programmatically-focused elements unstyled, and the user wondering where focus went.

We'd been able to work around this somewhat - but as I continue upgrading, even minor versions introduce breaking changes in this regard.

Is there any way to add an option like `focusRing: 'auto-plus'`, or to simply include programmatically-focused elements in `focusRing: 'auto'`?

I tried to investigate the relevant code, but I haven't been able to locate it in the Mantine repos yet.

Related to #2927.

**Comment by rtivital** [comment]
You can find focus styles logic here – https://github.com/mantinedev/mantine/blob/master/src/mantine-styles/src/theme/functions/fns/focus-styles/focus-styles.ts

I've explained in the issue how it works. We will not add the feature described in this discussion because `:focus-visible` selector behavior is different across browsers. For example, you will see focus ring when you programmatically focus an element in Safari and will not see it in Chrome.
